engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/sink/AvroSinkValueParameter.scala

```diff
@@ -0,0 +1,83 @@
+package pl.touk.nussknacker.engine.avro.sink
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.definition.Parameter
+import pl.touk.nussknacker.engine.api.typed
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypedObjectTypingResult, TypedUnion, TypingResult}
+import pl.touk.nussknacker.engine.avro.KafkaAvroBaseTransformer.{SchemaVersionParamName, SinkKeyParamName, SinkValidationModeParameterName, SinkValueParamName, TopicParamName}
+import pl.touk.nussknacker.engine.avro.sink.AvroSinkValueParameter.FieldName
+import pl.touk.nussknacker.engine.definition.parameter.editor.ParameterTypeEditorDeterminer
+
+private[sink] case object AvroSinkValueParameter {
+
+  type FieldName = String
+
+  private val restrictedParamNames = Set(SchemaVersionParamName, SinkKeyParamName, SinkValidationModeParameterName, TopicParamName)
+
+  def apply(typing: TypingResult)(implicit nodeId: NodeId): Validated[ProcessCompilationError, AvroSinkValueParameter] =
+    toSinkValueParameter(typing, paramName = None, isTopLevel = true)
+
+  private def toSinkValueParameter(typing: TypingResult, paramName: Option[String], isTopLevel: Boolean)
+                                  (implicit nodeId: NodeId): Validated[ProcessCompilationError, AvroSinkValueParameter] =
+    typing match {
+      case typed.typing.Unknown =>
+        Invalid(CustomNodeError(nodeId.id, "Cannot determine typing for provided schema", None))
+
+      case typedObject: TypedObjectTypingResult if containsRestrictedNames(typedObject) =>
+        Invalid(CustomNodeError(nodeId.id, s"""Record field name is restricted. Restricted names are ${restrictedParamNames.mkString(", ")}""", None))
+
+      /* kafka-avro-serializer does not support Array at top level
+         [https://github.com/confluentinc/schema-registry/issues/1298] */
+      case TypedClass(clazz, _) if isTopLevel && clazz == classOf[java.util.List[_]] =>
```

Comment:
Should it be handled here? We already have TypingResult from schema from schema regitry, so maybe we should just assume it's sth schema registry can handle?

Reply:
:+1:  done
https://github.com/TouK/nussknacker/pull/1405/commits/88e3672307aa1eedbbf4819b6fc8dd86cd63c1f6